libs/utils/plot_utils.py

```diff
@@ -0,0 +1,138 @@
+                 drawstyle="default", ax=None, title=""):
+    """
+    Generic trace plotting method
+
+    The values in the column 'name_field' will be used as discriminant
+
+    Let's consider a df with these columns:
+
+    | time | cpu | load_avg | util_avg |
+    ====================================
+    |  42  |  2  |   1812   |   400    |
+    ------------------------------------
+    |  43  |  0  |   1337   |   290    |
+    ------------------------------------
+    |  ..  | ... |    ..    |    ..    |
+
+    To plot the 'util_avg' value of CPU2, the function would be used like so:
+    plot_generic(df, name_field='cpu', names=[2], fields=['util_avg'])
+
+    CPUs could be compared by using:
+    plot_generic(df, name_field='cpu', names=[2, 3], fields=['util_avg'])
+
+    :param dfr: Trace dataframe
+    :type dfr: `pandas.DataFrame`
+
+    :param name_field: Column name to organize data set
+    :type name_field: string
+
+    :param names: Values of 'name_field' to use
+    :type names: list of str
+
+    :param fields: Name of columns whose data will be plotted
+    :type fields: list of str
+
+    :param prettify_name: name_field user-friendly stringify function
+    :type prettify_name: callable[str]
+
+    :param width: The width of the plot
+    :type width: int
+
+    :param height: The height of the plot
+    :type height: int
+
+    :param xlim: The xlim setting of the plot.
+    :type xlim: tuple(int, int)
+
+    :param drawstyle: The drawstyle setting of the plot
+    :type drawstyle: str
+    """
+
+    if prettify_name is None:
+        def prettify_name(name): return '{}={}'.format(name_field, name)
+
+    if names is None:
+        # Find available names
+        names = dfr[name_field].unique().tolist()
+    else:
+        # Filter out unwanted names
+        dfr = dfr[
+            dfr[name_field].isin(names)
+        ]
+
+        if len(dfr) == 0:
+            raise ValueError("No trace data for any of {}".format(
+                map(prettify_name, names)
+            ))
+
+    if fields is None:
+        # Find available fields
+        fields = dfr.columns.tolist()
+        fields.remove(name_field)
+    else:
+        # Filter out unwanted fields
+        try:
+            dfr = dfr[fields + [name_field]]
+        except KeyError as err:
+            raise ValueError("Invalid 'fields' parameter value : {}"
+                             .format(err.message)
+            )
+
+    setup_plot = False
+    if ax is None:
+        fig, ax = pre_plot_setup(width, height)
+        setup_plot = True
+
+    for name in names:
+        cols = []
+        for field in fields:
+            cols.append('{} {}'.format(prettify_name(name), field))
+
+        plot_dfr = dfr[dfr[name_field] == name][fields]
+        plot_dfr.columns = cols
+        plot_dfr.plot(ax=ax, drawstyle=drawstyle)
+
+    if setup_plot:
+        ax.set_title(title)
+
+    if xlim is not None:
+        ax.set_xlim(xlim)
```

Comment:
Maybe this could be added to the AnalysisModule itself, so that this can directly use `trace.x_min` & `trace.x_max` ?